packages/venia-concept/src/components/MiniCart/body.js

```diff
@@ -0,0 +1,94 @@
+import React, { useCallback } from 'react';
+import { array, bool, func, object, shape, string } from 'prop-types';
+
+import { mergeClasses } from 'src/classify';
+import { loadingIndicator } from 'src/components/LoadingIndicator';
+import getCurrencyCode from 'src/util/getCurrencyCode';
+
+import defaultClasses from './body.css';
+import EditItem from './editItem';
+import EmptyMiniCartBody from './emptyMiniCartBody';
+import ProductList from './productList';
+
+const Body = props => {
+    // Props.
+    const {
+        beginEditItem,
+        cart,
+        closeDrawer,
+        endEditItem,
+        isCartEmpty,
+        removeItemFromCart,
+        updateItemInCart
+    } = props;
+
+    // Members.
+    const classes = mergeClasses(defaultClasses, props.classes);
+    const currencyCode = getCurrencyCode(cart);
+    const { editItem, isLoading, isEditingItem, isUpdatingItem } = cart;
+
+    // Callbacks.
+    const handleEditItem = useCallback(
+        item => {
+            beginEditItem(item);
+        },
+        [beginEditItem]
+    );
+
+    // Render.
+    if (isLoading) {
+        return loadingIndicator;
+    }
+
+    if (isCartEmpty) {
+        return <EmptyMiniCartBody closeDrawer={closeDrawer} />;
+    }
+
+    if (isEditingItem) {
+        return (
+            <EditItem
+                currencyCode={currencyCode}
+                endEditItem={endEditItem}
+                isUpdatingItem={isUpdatingItem}
+                item={editItem}
+                updateItemInCart={updateItemInCart}
+            />
+        );
+    }
+
+    return (
+        <div className={classes.root}>
+            <ProductList
+                beginEditItem={handleEditItem}
+                cart={cart}
```

Comment:
You already did some of the work to not have to drill this down to this component. `ProductList` only needs `items` and `currencyCode`, in fact it did so before the refactor so I think you can just undo that and pass the specific props here.

I think you even mentioned avoiding passing the entire `cart` as a prop so I would just check that you aren't doing that unnecessarily elsewhere.

Reply:
Great call, this simplified things a lot. Resolved in https://github.com/magento-research/pwa-studio/pull/1266/commits/4a15cba0b10219ed4f92d33b897cd2826944f93d.

Unfortunately we still have to pass the entire `cart` from `MiniCart` to `Footer` because `Checkout` needs it. But I was able to simplify all the other places 👍